Cython/Utility/Coroutine.c

```diff
@@ -879,30 +956,58 @@ PyObject *__Pyx_PyGen_Send(PyGenObject *gen, PyObject *arg) {
 }
 #endif
 
-static CYTHON_INLINE
-PyObject *__Pyx_Coroutine_FinishDelegation(__pyx_CoroutineObject *gen) {
-    PyObject *ret;
+static CYTHON_INLINE __Pyx_PySendResult
+__Pyx_Coroutine_FinishDelegation(__pyx_CoroutineObject *gen, PyObject** retval) {
+    __Pyx_PySendResult result;
     PyObject *val = NULL;
     __Pyx_Coroutine_Undelegate(gen);
     __Pyx_PyGen__FetchStopIterationValue(__Pyx_PyThreadState_Current, &val);
     // val == NULL on failure => pass on exception
-    ret = __Pyx_Coroutine_SendEx(gen, val, 0);
+    result = __Pyx_Coroutine_SendEx(gen, val, retval, 0);
     Py_XDECREF(val);
-    return ret;
+    return result;
+}
+
+static __Pyx_PySendResult
+__Pyx_Coroutine_SendToDelegate(__pyx_CoroutineObject *gen, __Pyx_pyiter_sendfunc gen_am_send, PyObject *value, PyObject **retval) {
+    PyObject *ret;
+    __Pyx_PySendResult result;
+    // we assume that gen->yieldfrom cannot change as long as 'gen->is_running' is set => no safety INCREF()
+    gen->is_running = 1;
```

Comment:
Not for this PR, but I believe that `is_running` needs to be handled a little more carefully in a free-threading world.

Reply:
Probably, yes. I think there were changes in this area in CPython. Large parts of the code in this PR were copied from CPython.